src/sklearn_compat/_sklearn_compat.py

```diff
@@ -27,26 +28,6 @@
 ########################################################################################
```

Comment:
At the time, it was because the `ParameterMixin` itself is not part of any scikit-learn code.

> I just don't like to use sklearn version as a proxy for what I need to check (imports)

In general, I get this point but here the full library is based on version. I find it much more explicit and in line with the documentation then. Dropping code block of code is also easier.

Reply:
that's what comments are for 😁

Reply:
and this will bite us when we start supporting our nightly releases

Reply:
Not really because we are using the `base_version`, e.g.:

```python
parse_version(parse_version("1.5.dev0").base_version) >= parse_version("1.5")
```